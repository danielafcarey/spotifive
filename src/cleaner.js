const getAccessToken = (url) => {
  const splitUrl = url.split('=');

  return splitUrl[1];
};

const cleanUserData = (userData, playlists) => {
  const spotifive = playlists.find(playlist => {
    return playlist.name === 'Spotifive';
  });

  return {
    userId: userData.id,
    name: userData.display_name,
    image: userData.images.length > 0 ? userData.images[0].url : null,
    spotifive: {
      spotifiveId: spotifive ? spotifive.id : null,
      link: spotifive ? spotifive.external_urls.spotify : null
    }
  };
};

const cleanSearchResults = (searchResults) => {
  const cleanedSearchResults = searchResults.artists.items.map(result => {
    const cleanResult = {
      name: result.name,
      id: result.id,
      image: result.images.length ? result.images[0].url : null
    };
    return cleanResult;
  });

  return cleanedSearchResults;
};

const cleanArtistData = (artistData, topTracks) => {
  const cleanedArtistData = {
    name: artistData.name,
    id: artistData.id,
    image: artistData.images.length ? artistData.images[0].url : null,
    topTracks: cleanTopTracks(topTracks.tracks)
  };

  return cleanedArtistData;
};

const cleanTopTracks = (topTracks) => {
  const cleanedTopTracks = topTracks.map(track => {
    return {
      name: track.name,
      uri: track.uri
    };
  }).splice(0, 5);

  return cleanedTopTracks;
};

export {
  getAccessToken,
  cleanUserData,
  cleanSearchResults,
  cleanArtistData,
  cleanTopTracks
};
