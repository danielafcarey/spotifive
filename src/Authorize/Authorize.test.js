import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import * as cleaner from '../cleaner';
import {
  Authorize,
  mapDispatchToProps
} from './Authorize';

import {
  updateAccessToken
} from '../actions';
cleaner.getAccessToken = jest.fn().mockImplementation(() => 'cleanedAccessToken')



describe('Authorize', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      updateAccessToken: jest.fn(),
    }
    window.location = { hash: 'string' }
    wrapper = shallow(<Authorize { ...mockProps }/>);
  }) 

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('componentDidMount', () => {

    it('calls updateAccessToken with the correct argumenst', () => {
      wrapper = shallow(<Authorize { ...mockProps }/>);
      const expected = cleaner.getAccessToken();

      expect(wrapper.instance().props.updateAccessToken).toHaveBeenCalledWith(expected);
    })
  })

  describe('mapDispatchToProps', () => {

    it('should call dispatch with the correct arguments', () => {
      const dispatch = jest.fn();
      const mockAccessToken = 'token';
      const mockAction = {
        type: 'UPDATE_ACCESS_TOKEN',
        accessToken: mockAccessToken
      }
      const mappedProps = mapDispatchToProps(dispatch);

      mappedProps.updateAccessToken(mockAction.accessToken);

      expect(dispatch).toHaveBeenCalledWith(mockAction);
    })
  })

})

