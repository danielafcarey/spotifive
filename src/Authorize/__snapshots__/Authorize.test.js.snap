// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorize matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Authorize
    accessToken=""
    submitUpdateUser={[Function]}
    updateAccessToken={[Function]}
    user={
        Object {
            "userId": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "authorize",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "authorize",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
